app/models/grda_warehouse/warehouse_clients_processed.rb

```diff
@@ -810,6 +812,77 @@ def vispdat_priority_score(client_id)
       end
     end
 
+    def last_intentional_contacts(client_id)
+      open_enrollments = GrdaWarehouse::ServiceHistoryEnrollment.entry.ongoing.
+        where(client_id: @client_ids)
+
+      @services ||= {}.tap do |h|
+        open_enrollments.
+          joins(enrollment: [:project, :services]).
+          group(she_t[:client_id], p_t[:project_name], p_t[:id]).maximum(s_t[:date_provided]).
+          map do |(service_client_id, project_name, project_id), date|
+            h[service_client_id] ||= {}
+            h[service_client_id][project_id] ||= {
+              date: date,
+              project_name: project_name,
```

Comment:
Let's remove `project_name` except when we don't have an id.  That will enforce confidentiality (and save space!)